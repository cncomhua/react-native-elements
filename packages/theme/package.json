{
  "name": "@react-native-elements/theme",
  "version": "1.0.0",
  "description": "",
  "main": "src/index.tsx",
  "react-native": "src/index.tsx",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "peerDependencies": {
    "@react-native-element/base": "*"
  },
  "author": "",
  "license": "ISC"
}
