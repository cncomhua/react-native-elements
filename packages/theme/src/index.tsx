import React from "react";
import { Pressable, Text, StyleSheet } from "react-native";
// import { FAB } from "./src/FAB/FAB";

const s = StyleSheet.create({
  button: {
    backgroundColor: "#f00",
    padding: 10,
    margin: 10,
    borderRadius: 10,
  },
});
export const ButtonComp = () => {
  return (
    <>
      {/* <Tab value={0}>
        <TabItem title="Tab 1" />
        <TabItem title="Tab 2" />
        <TabItem title="Tab 3" />
      </Tab> */}
      {/* 
      <FAB title={"FAB"} />
      <Button title="working" /> */}
      <Pressable style={s.button}>
        <Text>Yes Working...</Text>
      </Pressable>
    </>
  );
};

// export * from "./src";
