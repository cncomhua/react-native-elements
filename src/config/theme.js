// For testing purposes only
import colors from './colors';

export default {
  colors,
};
