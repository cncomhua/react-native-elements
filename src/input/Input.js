import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text, View, TextInput, Dimensions } from 'react-native';
import ViewPropTypes from '../config/ViewPropTypes';
const SCREEN_WIDTH = Dimensions.get('window').width;

class Input extends Component {
  focus() {
    this.input.focus();
  }

  blur() {
    this.input.blur();
  }

  clear() {
    this.input.clear();
  }

  render() {
    const {
      containerStyle,
      icon,
      iconContainerStyle,
      inputStyle,
      displayError,
      errorStyle,
      errorMessage,
      ...attributes
    } = this.props;

    const styles = this.context.theme.input;

    return (
      <View>
        <View
          style={[
            styles.container,
            { width: SCREEN_WIDTH - 100, height: 40 },
            containerStyle,
          ]}
        >
          {icon &&
            <View
              style={[styles.iconContainer, { height: 40 }, iconContainerStyle]}
            >
              {icon}
            </View>}
          <TextInput
            ref={input => (this.input = input)}
            underlineColorAndroid="transparent"
            style={[
              styles.input,
              { width: SCREEN_WIDTH - 100, height: 40 },
              inputStyle,
            ]}
            {...attributes}
          />
        </View>
        {displayError &&
          <Text style={[styles.error, errorStyle && errorStyle]}>
            {errorMessage || 'Error!'}
          </Text>}
      </View>
    );
  }
}

Input.propTypes = {
  containerStyle: ViewPropTypes.style,
  icon: PropTypes.object,
  iconContainerStyle: ViewPropTypes.style,
  inputStyle: PropTypes.object,
  displayError: PropTypes.bool,
  errorStyle: PropTypes.object,
  errorMessage: PropTypes.string,
};

Input.contextTypes = {
  theme: PropTypes.object.isRequired,
};

export default Input;
