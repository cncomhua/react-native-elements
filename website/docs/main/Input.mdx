---
id: input
title: Input
slug: /input
---

import Usage from './usage/Input/Input.mdx';

import Props from './props/Input.mdx';

Inputs allow users to enter text into a UI. They typically appear in forms and
dialogs.

<div className="component-preview component-preview--grid">
  <figure>
    <img src="/img/input/input--placeholder.png" alt="Input with placeholder" />
    <figcaption>Placeholder</figcaption>
  </figure>
  <figure>
    <img src="/img/input/input--label.png" alt="Input with Label and icons" />
    <figcaption>Label with Icons</figcaption>
  </figure>
</div>

## Usage

<Usage />

---

## Props

<Props />
