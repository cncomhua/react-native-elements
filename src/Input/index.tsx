import { withTheme } from '../config';
import { Input, InputProps } from './Input';

export { Input };
export type { InputProps };
/**
 * RNE Input
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/input)
 */
export default withTheme(Input, 'Input');
