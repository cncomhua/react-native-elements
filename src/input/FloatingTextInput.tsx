import React from 'react';
import {
  View,
  Animated,
  StyleSheet,
  TextInput,
  TextStyle,
  StyleProp,
  TextInputProps,
} from 'react-native';
import { RneFunctionComponent } from '../helpers';
import { withTheme } from '../config';

export type FloatingInputProps = TextInputProps & {
  attrName?: string;
  title?: string;
  value?: string;
  update?: (attrName: string, updatedValue: string) => any;
  titleActiveSize?: number;
  titleInActiveSize?: number;
  titleActiveColor?: string;
  titleInActiveColor?: string;
  textInputStyles?: StyleProp<TextStyle>;
  otherTextInputProps?: {};
};

const FloatingTextInput: RneFunctionComponent<FloatingInputProps> = ({
  keyboardType = 'default',
  titleActiveSize = 14,
  titleInActiveSize = 19,
  titleActiveColor = 'blue',
  titleInActiveColor = 'dimgrey',
  textInputStyles = {},
  ...props
}) => {
  const [isFieldActive, setIsFieldActive] = React.useState(false);
  const { current: position } = React.useRef(
    new Animated.Value(isFieldActive ? 1 : 0)
  );

  const handleFocus = () => {
    if (!isFieldActive) {
      setIsFieldActive(true);
      Animated.timing(position, {
        toValue: 1,
        duration: 150,
        useNativeDriver: true,
      }).start();
    }
  };

  const handleBlur = () => {
    if (isFieldActive && !props.value) {
      setIsFieldActive(false);
      Animated.timing(position, {
        toValue: 1,
        duration: 150,
        useNativeDriver: true,
      }).start();
    }
  };

  const onChangeText = (updatedValue) => {
    const { attrName, update } = props;
    update(attrName, updatedValue);
  };

  const returnAnimatedTitleStyle = () => {
    return {
      top: position.interpolate({
        inputRange: [0, 1],
        outputRange: [14, 0],
      }),
      fontSize: isFieldActive ? titleActiveSize : titleInActiveSize,
      color: isFieldActive ? titleActiveColor : titleInActiveColor,
    };
  };

  return (
    <View style={styles.container}>
      <Animated.Text style={[styles.titleStyles, returnAnimatedTitleStyle()]}>
        {props.title}
      </Animated.Text>

      <TextInput
        value={props.value}
        style={[styles.textInput, textInputStyles]}
        underlineColorAndroid="transparent"
        onFocus={handleFocus}
        onBlur={handleBlur}
        onChangeText={onChangeText}
        keyboardType={keyboardType}
        {...props.otherTextInputProps}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    borderRadius: 3,
    borderStyle: 'solid',
    borderWidth: 0.5,
    height: 50,
    marginVertical: 4,
  },
  textInput: {
    fontSize: 15,
    marginTop: 5,
    fontFamily: 'Avenir-Medium',
    color: 'black',
  },
  titleStyles: {
    position: 'absolute',
    fontFamily: 'Avenir-Medium',
    left: 3,
  },
});

export { FloatingTextInput };
export default withTheme(FloatingTextInput, 'FloatingInput');
