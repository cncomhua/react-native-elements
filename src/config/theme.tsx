import { ViewProps } from 'react-native';
import { SwipeRatingProps } from '../Rating';
import { TapRatingProps } from '../AirbnbRating';
import { AvatarProps, AccessoryProps } from '../Avatar';
import { BadgeProps } from '../Badge';
import { BottomSheetProps } from '../BottomSheet';
import { ButtonProps } from '../Button';
import { ButtonGroupProps } from '../ButtonGroup';
import { CardProps } from '../card/Card';
import { CheckBoxProps } from '../Checkbox';
import { DividerProps } from '../Divider';
import { HeaderProps } from '../Header';
import { IconProps } from '../Icon';
import { ImageProps } from '../Image';
import { InputProps } from '../Input';
import { ListItemProps } from '../ListItem';
import { ListItemAccordionProps } from '../ListItem';
import { OverlayProps } from '../Overlay';
import { PricingCardProps } from '../PricingCard';
import { SearchBarProps } from '../searchbar/SearchBar';
import { SliderProps } from '../Slider';
import { SocialIconProps } from '../SocialIcon';
import { TextProps } from '../Text';
import { TileProps } from '../Tile';
import { TooltipProps } from '../Tooltip';
import { SwitchProps } from '../Switch';
import { TabItemProps, TabProps } from '../tab/Tab';
import { TabViewProps } from '../tab/TabView';
import { FABProps } from '../FAB';
import { SpeedDialProps, SpeedDialActionProps } from '../SpeedDial';
import { LinearProgressProps } from '../LinearProgress';
import { ChipProps } from '../Chip';
import {
  DialogActionsProps,
  DialogLoadingProps,
  DialogProps,
  DialogTitleProps,
} from '../Dialog';

import colors, { Colors } from './colors';

export default {
  colors,
};

type RecursivePartial<T> = { [P in keyof T]?: RecursivePartial<T[P]> };

export interface FullTheme {
  Avatar: Partial<AvatarProps>;
  AvatarAccessory: Partial<AccessoryProps>;
  Badge: Partial<BadgeProps>;
  BottomSheet: Partial<BottomSheetProps>;
  Button: Partial<ButtonProps>;
  ButtonGroup: Partial<ButtonGroupProps>;
  Chip: Partial<ChipProps>;
  Card: Partial<CardProps>;
  CardDivider: Partial<DividerProps>;
  CardFeaturedSubtitle: Partial<TextProps>;
  CardFeaturedTitle: Partial<TextProps>;
  CardImage: Partial<ImageProps>;
  CardTitle: Partial<TextProps>;
  CheckBox: Partial<CheckBoxProps>;
  Divider: Partial<DividerProps>;
  Dialog: Partial<DialogProps>;
  DialogActions: Partial<DialogActionsProps>;
  DialogButton: Partial<ButtonProps>;
  DialogLoading: Partial<DialogLoadingProps>;
  DialogTitle: Partial<DialogTitleProps>;
  Header: Partial<HeaderProps>;
  Icon: Partial<IconProps>;
  Image: Partial<ImageProps>;
  Input: Partial<InputProps>;
  ListItem: Partial<ListItemProps>;
  ListItemAccordion: Partial<ListItemAccordionProps>;
  ListItemButtonGroup: Partial<ButtonGroupProps>;
  ListItemCheckBox: Partial<CheckBoxProps>;
  ListItemContent: Partial<ViewProps>;
  ListItemChevron: Partial<IconProps>;
  ListItemInput: Partial<InputProps>;
  ListItemSubtitle: Partial<TextProps>;
  ListItemTitle: Partial<TextProps>;
  Overlay: Partial<OverlayProps>;
  PricingCard: Partial<PricingCardProps>;
  Rating: Partial<SwipeRatingProps>;
  AirbnbRating: Partial<TapRatingProps>;
  SearchBar: Partial<SearchBarProps>;
  Slider: Partial<SliderProps>;
  SocialIcon: Partial<SocialIconProps>;
  Text: Partial<TextProps>;
  Tile: Partial<TileProps>;
  Switch: Partial<SwitchProps>;
  Tooltip: Partial<TooltipProps>;
  colors: RecursivePartial<Colors>;
  Tab: Partial<TabProps>;
  TabItem: Partial<TabItemProps>;
  TabView: Partial<TabViewProps>;
  TabViewItem: Partial<ViewProps>;
  FAB: Partial<FABProps>;
  SpeedDial: Partial<SpeedDialProps>;
  SpeedDialAction: Partial<SpeedDialActionProps>;
  LinearProgress: Partial<LinearProgressProps>;
}

export type Theme<T = {}> = Partial<FullTheme> & T;

export type UpdateTheme = (updates: RecursivePartial<FullTheme>) => void;

export type ReplaceTheme = (updates: RecursivePartial<FullTheme>) => void;
