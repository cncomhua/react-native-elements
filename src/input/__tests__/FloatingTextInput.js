import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import theme from '../../config/theme';
import { FloatingTextInput } from '../FloatingTextInput';
import { Input } from '../Input';

describe('Input component', () => {
  beforeAll(() => {
    // useNativeDriver isn't available in jest, so just silencing the warning
    global.console.warn = () => null;
  });

  it('should match snapshot', () => {
    const component = shallow(<FloatingTextInput theme={theme} />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  describe('Props', () => {
    it('attrName', () => {
      const component = shallow(
        <FloatingTextInput theme={theme} attrName="My Attribute Name" />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('title', () => {
      const component = shallow(
        <FloatingTextInput theme={theme} title="My Title Name" />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('value', () => {
      const component = shallow(
        <FloatingTextInput theme={theme} title="My Title Name" />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('titleActiveSize', () => {
      const component = shallow(<FloatingTextInput titleActiveSize={14} />);
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('titleInActiveSize', () => {
      const component = shallow(<FloatingTextInput titleInActiveSize={19} />);
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('titleActiveColor', () => {
      const component = shallow(
        <FloatingTextInput theme={theme} titleActiveColor="blue" />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('titleInActiveColor', () => {
      const component = shallow(
        <FloatingTextInput theme={theme} titleInActiveColor="dimgrey" />
      );
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });
  });

  describe('Instance methods', () => {
    it('should focus the input', () => {
      const focus = jest.fn();
      const component = shallow(<Input theme={theme} />);
      const instance = component.instance();
      // Refs not available in shallow render
      instance.input = {
        focus,
      };
      instance.focus();
      expect(focus).toHaveBeenCalledTimes(1);
    });

    it('should blur the input', () => {
      const blur = jest.fn();
      const component = shallow(<Input theme={theme} />);
      const instance = component.instance();
      // Refs not available in shallow render
      instance.input = {
        blur,
      };
      instance.blur();
      expect(blur).toHaveBeenCalledTimes(1);
    });
  });

  // it('should apply values from theme', () => {
  //   const testTheme = {
  //     FloatingInput: {
  //       placeholder: 'Enter text',
  //     },
  //   };
  //   const component = create(
  //     <ThemeProvider theme={testTheme}>
  //       <ThemedFloatingTextInput />
  //     </ThemeProvider>
  //   );
  //   expect(component.root.findByType(TextInput).props.placeholder).toBe(
  //     'Enter text'
  //   );
  //   expect(component.toJSON()).toMatchSnapshot();
  // });
});
