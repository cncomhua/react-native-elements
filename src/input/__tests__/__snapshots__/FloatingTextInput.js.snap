// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Props attrName 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "height": 50,
      "marginVertical": 4,
      "width": "100%",
    }
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Medium",
          "left": 3,
          "position": "absolute",
        },
        Object {
          "color": "dimgrey",
          "fontSize": 19,
          "top": 14,
        },
      ]
    }
  />
  <Component
    allowFontScaling={true}
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "Avenir-Medium",
          "fontSize": 15,
          "marginTop": 5,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Input component Props title 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "height": 50,
      "marginVertical": 4,
      "width": "100%",
    }
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Medium",
          "left": 3,
          "position": "absolute",
        },
        Object {
          "color": "dimgrey",
          "fontSize": 19,
          "top": 14,
        },
      ]
    }
  >
    My Title Name
  </ForwardRef(AnimatedComponentWrapper)>
  <Component
    allowFontScaling={true}
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "Avenir-Medium",
          "fontSize": 15,
          "marginTop": 5,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Input component Props titleActiveColor 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "height": 50,
      "marginVertical": 4,
      "width": "100%",
    }
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Medium",
          "left": 3,
          "position": "absolute",
        },
        Object {
          "color": "dimgrey",
          "fontSize": 19,
          "top": 14,
        },
      ]
    }
  />
  <Component
    allowFontScaling={true}
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "Avenir-Medium",
          "fontSize": 15,
          "marginTop": 5,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Input component Props titleActiveSize 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "height": 50,
      "marginVertical": 4,
      "width": "100%",
    }
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Medium",
          "left": 3,
          "position": "absolute",
        },
        Object {
          "color": "dimgrey",
          "fontSize": 19,
          "top": 14,
        },
      ]
    }
  />
  <Component
    allowFontScaling={true}
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "Avenir-Medium",
          "fontSize": 15,
          "marginTop": 5,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Input component Props titleInActiveColor 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "height": 50,
      "marginVertical": 4,
      "width": "100%",
    }
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Medium",
          "left": 3,
          "position": "absolute",
        },
        Object {
          "color": "dimgrey",
          "fontSize": 19,
          "top": 14,
        },
      ]
    }
  />
  <Component
    allowFontScaling={true}
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "Avenir-Medium",
          "fontSize": 15,
          "marginTop": 5,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Input component Props titleInActiveSize 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "height": 50,
      "marginVertical": 4,
      "width": "100%",
    }
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Medium",
          "left": 3,
          "position": "absolute",
        },
        Object {
          "color": "dimgrey",
          "fontSize": 19,
          "top": 14,
        },
      ]
    }
  />
  <Component
    allowFontScaling={true}
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "Avenir-Medium",
          "fontSize": 15,
          "marginTop": 5,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Input component Props value 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "height": 50,
      "marginVertical": 4,
      "width": "100%",
    }
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Medium",
          "left": 3,
          "position": "absolute",
        },
        Object {
          "color": "dimgrey",
          "fontSize": 19,
          "top": 14,
        },
      ]
    }
  >
    My Title Name
  </ForwardRef(AnimatedComponentWrapper)>
  <Component
    allowFontScaling={true}
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "Avenir-Medium",
          "fontSize": 15,
          "marginTop": 5,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Input component should match snapshot 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "height": 50,
      "marginVertical": 4,
      "width": "100%",
    }
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Medium",
          "left": 3,
          "position": "absolute",
        },
        Object {
          "color": "dimgrey",
          "fontSize": 19,
          "top": 14,
        },
      ]
    }
  />
  <Component
    allowFontScaling={true}
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "Avenir-Medium",
          "fontSize": 15,
          "marginTop": 5,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
